src/IECoreDelight/Renderer.cpp

```diff
@@ -1028,6 +1046,176 @@ class DelightObject : public IECoreScenePreview::Renderer::ObjectInterface
 
 } // namespace
 
+namespace
+{
+
+const InternedString g_radiusParameterName = "radius";
+
+class DelightLight : public IECoreScenePreview::Renderer::ObjectInterface
+{
+
+	public :
+
+		DelightLight( NSIContext_t context, const std::string &name, InstanceCachePtr instanceCache, DelightHandle::Ownership ownership )
+			: m_transformHandle( context, name, ownership, "transform", {} ), m_instanceCache( instanceCache ), m_usdLightType( USDLightType::None), m_ownership( ownership ), m_haveTransform( false )
+		{
+			NSIConnect(
+				m_transformHandle.context(),
+				m_transformHandle.name(), "",
+				NSI_SCENE_ROOT, "objects",
+				0, nullptr
+			);
+		}
+
+		void transform( const Imath::M44f &transform ) override
+		{
+			if( transform == M44f() && !m_haveTransform )
+			{
+				return;
+			}
+
+			M44d m( transform );
+			NSIParam_t param = {
+				"transformationmatrix",
+				m.getValue(),
+				NSITypeDoubleMatrix,
+				0, 1, // array length, count
+				0 // flags
+			};
+			NSISetAttribute( m_transformHandle.context(), m_transformHandle.name(), 1, &param );
+
+			m_haveTransform = true;
+		}
+
+		void transform( const std::vector<Imath::M44f> &samples, const std::vector<float> &times ) override
+		{
+			if( m_haveTransform )
+			{
+				NSIDeleteAttribute( m_transformHandle.context(), m_transformHandle.name(), "transformationmatrix" );
+			}
+
+			for( size_t i = 0, e = samples.size(); i < e; ++i )
+			{
+				M44d m( samples[i] );
+				NSIParam_t param = {
+					"transformationmatrix",
+					m.getValue(),
+					NSITypeDoubleMatrix,
+					0, 1, // array length, count
+					0 // flags
+				};
+				NSISetAttributeAtTime( m_transformHandle.context(), m_transformHandle.name(), times[i], i, &param );
+			}
+
+			m_haveTransform = true;
+		}
```

Comment:
Is there a good reason to duplicate this from DelightObject? Can we share the code by deriving from DelightObject or a new DelightObjectBase class?

Reply:
No good reason, moving to `DelightObjectBase` worked well, it does everything `DelightObject` was doing, and now `DelightLight` is much simpler, just overriding `attributes()`.